### PR TITLE
Add AWS-specific Ansible roles to our Packer configuration

### LINKED ISSUE #397: Add AWS CloudWatch agent to CyHy AMIs
# 🚀 Feature Proposal #

Add the AWS CloudWatch Agent to the CyHy AMIs, particularly the AMI that hosts the CyHy commander.

## Motivation ##

The CloudWatch Agent is required for the changes in cisagov/cyhy_amis#394, cisagov/cyhy-core#67, and cisagov/cyhy_amis#396 to function as intended.

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

The core goal of this pull request is to add the following Ansible roles to the AMIs defined in this project:

- [cisagov/ansible-role-amazon-ssm-agent](https://github.com/cisagov/ansible-role-amazon-ssm-agent)
- [cisagov/ansible-role-chrony-aws](https://github.com/cisagov/ansible-role-chrony-aws)
- [cisagov/ansible-role-cloudwatch-agent](https://github.com/cisagov/ansible-role-cloudwatch-agent)

As part of adding the CloudWatch and SSM Agents to the AMIs there are corresponding changes to the Terraform configurations to both bring the instance definitions more in line with modern projects like [cisagov/cool-assessment-terraform](https://github.com/cisagov/cool-assessment-terraform) and to provide appropriate IAM permissions for instances to leverage these Agents. This closes #397.

## 💭 Motivation and context ##

Installing the CloudWatch and SSM Agents will allow us to better monitor and troubleshoot our instances. The CloudWatch logs will allow us to create alerts to better detect problems like jobs failing or running out of disk space (this latter has happened frequently and gone unnoticed for periods of time). Additionally the SSM Agent will allow us to connect to instances even if there are problems with the bastions or the networking configuration.

Reorganizing the IAM configurations for instances will as a side effect make these more maintainable and mirror changes in #409 to refactor the Route53 resources in this project.

## 🧪 Testing ##

Automated tests pass. I have built AMIs and deployed instances using these changes and have noticed no problems. In fact I was able to troubleshoot an issue with `cloud-init` by connecting to an instance through SSM.

### Caveat ###

The `local_hostname` value instances are populating for their CloudWatch log groups are the private IPv4 based names. I'm unsure why the DNS-based hostname isn't populating.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated
      to reflect the changes in this PR.
- [x] All new and existing tests pass.

### CODE REVIEW

**Review by jsf9k — APPROVED** [review]
_I LOVE IT!!!_

File: terraform/bod_docker_iam.tf [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    actions = [
+      "lambda:InvokeFunction",
+    ]
+
+    # I should be able to use splat syntax here
+    resources = [
+      aws_lambda_function.lambdas[0].arn,
+      aws_lambda_function.lambdas[1].arn,
+      aws_lambda_function.lambdas[2].arn,
+    ]
```
Comment (dav3r) [inline terraform/bod_docker_iam.tf:42]: We can get rid of these hard-coded lambda counts:
```suggestion
    resources = [ for lambda in aws_lambda_function.lambdas : lambda.arn ]
```

File: terraform/bod_lambdas.tf [hunk 2]
```diff
@@ -1,23 +1,9 @@
...
-    actions = ["sts:AssumeRole"]
-
-    principals {
-      type        = "Service"
-      identifiers = ["lambda.amazonaws.com"]
-    }
-  }
-}
-
 # The roles we're creating for the lambda functions
```
Comment (dav3r) [inline terraform/bod_lambdas.tf]: Capitalization - there are probably a bunch of other similar cases if you feel like fixing them:
```suggestion
# The roles we're creating for the Lambda functions
```

File: terraform/fdi_lambda.tf [hunk 3]
```diff
@@ -1,23 +1,9 @@
...
-}
-
 # The role we're creating for the lambda function
 resource "aws_iam_role" "fdi_lambda_role" {
-  assume_role_policy = data.aws_iam_policy_document.fdi_lambda_assume_role_doc.json
+  name               = format("fdi_lambda_role_%s", local.production_workspace ? "production" : terraform.workspace)
+  assume_role_policy = data.aws_iam_policy_document.lambda_service_assume_role_doc.json
 }
 
 # IAM policy document that that allows some Cloudwatch permissions
```
Comment (dav3r) [inline terraform/fdi_lambda.tf]: Capitalization - there are probably a bunch of other similar cases if you feel like fixing them:

```suggestion
# IAM policy document that that allows some CloudWatch permissions
```

**Review by dav3r — APPROVED** [review]
Beautiful work!  🚀 
Please note my minor comments.

File: terraform/bod_docker_iam.tf (hunk 1 above)
Comment (mcdonnnj) [reply terraform/bod_docker_iam.tf:42]: @dav3r I have that taken care of as part of some other work ([improvement/refactor_bod_lambda_configuration](https://github.com/cisagov/cyhy_amis/tree/improvement/refactor_bod_lambda_configuration)). Do you mind if I leave that out of this PR?

File: terraform/bod_lambdas.tf (hunk 2 above)
Comment (mcdonnnj) [reply terraform/bod_lambdas.tf]: Please see [021f2d9](https://github.com/cisagov/cyhy_amis/pull/410/commits/021f2d999cfd1d358eeec475db917ee0507cd8e2).

File: terraform/fdi_lambda.tf (hunk 3 above)
Comment (mcdonnnj) [reply terraform/fdi_lambda.tf]: Please see [021f2d9](https://github.com/cisagov/cyhy_amis/pull/410/commits/021f2d999cfd1d358eeec475db917ee0507cd8e2).

File: terraform/bod_docker_iam.tf (hunk 1 above)
Comment (dav3r) [reply terraform/bod_docker_iam.tf:42]: 👍 No problem at all.

**Pull request merged**